### PR TITLE
JetBrains: Various fixes and support for 2023.1

### LINKED ISSUE #50414: JetBrains: Extension incompatible with IntelliJ 2023.1
- **Sourcegraph version:** 5.0
- **Platform information:** IntelliJ 2023.1 

```
Plugin 'Sourcegraph' (version '2.1.2') is not compatible with the current version of the IDE, because it requires build 223.* or older but the current build is IU-231.8109.175
```

image (4)

If you would like immediate help on this, please email support@sourcegraph.com (you can still create the issue, but there are no [SLAs](https://about.sourcegraph.com/support/) on issues like there are for support requests).

### PR DESCRIPTION
**Pull request by philipp-spiess**
Fixes #50414

Some quick UI fixes for the JetBrains extension and support for 2023.1.

## Test plan

Screenshot 2023-04-12 at 12 07 23
Screenshot 2023-04-12 at 12 07 27

## App preview:

- [Web](https://sg-web-ps-jetbrains-2023-1.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.04 kb) | 0.00% (+0.04 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/jetbrains-2023.1/compare-report.html;tab=live_object) for a full comparison between the commits 512d64a and 185c154 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by philipp-spiess** [comment]
I will wait a bit before releasing this. Want to see if I can find a repro for [#50080](https://github.com/sourcegraph/sourcegraph/issues/50080) and work in a fix for this too. I’m having issues with the docker container currently though

File: .github/workflows/jetbrains-tests.yml [hunk 1]
```diff
@@ -30,12 +30,12 @@ jobs:
...
         uses: gradle/wrapper-validation-action@v1.0.4
 
-      # Setup Java 11 environment for the next steps
+      # Setup Java 18 environment for the next steps
       - name: Setup Java
         uses: actions/setup-java@v2
         with:
           distribution: zulu
-          java-version: 11
+          java-version: 18
```
Comment (vdavid) [inline .github/workflows/jetbrains-tests.yml:38]: I'm afraid it won't be backward compatible (with pre-2023.x) if we upgrade to Java 18. Have you checked this?

**Review by vdavid — COMMENTED** [review]
Looks good to me so far! Let me know when I should do a review on its final state!

File: client/jetbrains/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 ### Added
 
+- Compatibility with IntelliJ 2023.1
+
 ### Changed
```
Comment (vdavid) [inline client/jetbrains/CHANGELOG.md:9]: I think you'll need to remove the empty headers, as per [this](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/jetbrains/CONTRIBUTING.md?L41)

**Review by vdavid — APPROVED** [review]
Looks good! Must have been so annoying to do all the updates and then revert them. :/ I think I went through this exact same process (bump versions incl. Java to v 18) with the last IDE version bump. Not sure where we could leave a note to ourselves not to fall into this trap again. (And I hope in 1–1.5 years we can drop support for old IDE versions and upgrade to Java 18)

**Pull request merged**